Update build on OpenBSD

hubslave opened the pull request:
~~DISREGARD THIS PULL REQUEST FOR THE MOMENT, sorry, I forgot that it included some non-cleaned-up-changes as well. Will look at them tomorrow. Or if you already know how the missing off64_t and byteswap.h headers should be handled, drop a comment or change the PR directly (can maintainers do that? I don't know, I did check some box to allow editing something.)~~
Okay, PR fixed now. In short:
* `argp-standalone` needs to be linked as `-largp`, not `-lARGP`
* `byteswap.h` and `off64_t` and `lseek64` only exists on glibc; preprocessor shims for these already exist for OS X, so expand the checks to cover the BSD family
* update `doc/openbsd.txt` to reflect these changes

jgarzik commented:
Mostly looks good -- good catch on configure.ac fix.

The include/ccoin/endian.h change is not portable, unfortunately, and that small part cannot be merged as-is.

hubslave commented:
The `include/ccoin/endian.h` and `src/brd.c` changes are now cleaned up to be portable and moved to a dedicated commit. While there I added preprocessor checks for the other BSDs as well, though I've only tested a build on OpenBSD.
